modules/data/text/text.js

```diff
@@ -0,0 +1,51 @@
+// ref https://developer.mozilla.org/en-US/docs/Web/API/TextEncoder
+export class TextEncoder {
+	constructor() {
+
+	}
+	get encoding() {
+		return 'utf-8';
+	}
+	encode(s) {
+		if (typeof s !== 'string') {
+			throw new TypeError(typeof s);
+		}
+		let arrayBuffer;
+		let bytes;
+		if (s.length === 0) {
+			arrayBuffer = undefined;
+			bytes = Uint8Array.from([]);
```

Comment:
Can `new Uint8Array()` to avoid `[]` alloc.

Reply:
Can I hoist the empty `Uint8Array` to the top? Or do I have to return a distinct object each time?

Reply:
I would make a fresh empty array each time to avoid creating more mutable global scope out of SES’s reach.